src/jaxsim/physics/algos/aba.py

```diff
@@ -115,46 +136,72 @@ def loop_body_pass1(carry: Pass1Carry, i: jtp.Int) -> Tuple[Pass1Carry, None]:
         pA_i = Cross.vx_star(v[i]) @ M[i] @ v[i] - i_Xf_W @ jnp.vstack(f_ext[i])
         pA = pA.at[i].set(pA_i)
 
-        return (i_X_λi, v, c, MA, pA, i_X_0), None
+        pR_i = Cross.vx_star(m_v[i]) @ IM[i] @ m_v[i] - K̅ᵥ[i] * m_v[i]
+        pR = pR.at[i].set(pR_i)
+
+        return (i_X_λi, v, c, m_v, m_c, MA, pA, pR, i_X_0), None
 
-    (i_X_λi, v, c, MA, pA, i_X_0), _ = jax.lax.scan(
+    (i_X_λi, v, c, m_v, m_c, MA, pA, pR, i_X_0), _ = jax.lax.scan(
         f=loop_body_pass1,
         init=pass_1_carry,
         xs=np.arange(start=1, stop=model.NB),
     )
 
     U = jnp.zeros_like(S)
+    m_U = jnp.zeros_like(S)
     d = jnp.zeros(shape=(model.NB, 1))
     u = jnp.zeros(shape=(model.NB, 1))
+    m_u = jnp.zeros(shape=(model.NB, 1))
 
     Pass2Carry = Tuple[
         jtp.MatrixJax,
         jtp.MatrixJax,
         jtp.MatrixJax,
         jtp.MatrixJax,
         jtp.MatrixJax,
+        jtp.MatrixJax,
+        jtp.MatrixJax,
     ]
 
-    pass_2_carry = (U, d, u, MA, pA)
+    pass_2_carry = (U, m_U, d, u, m_u, MA, pA)
 
     # Pass 2
     def loop_body_pass2(carry: Pass2Carry, i: jtp.Int) -> Tuple[Pass2Carry, None]:
         ii = i - 1
-        U, d, u, MA, pA = carry
+        U, m_U, d, u, m_u, MA, pA = carry
 
         # Compute intermediate results
+        u_i = tau[ii] - S[i].T @ pA[i] if tau.size != 0 else -S[i].T @ pA[i]
+        u = u.at[i].set(u_i.squeeze())
+
+        has_motors = Γ[i] != 1
```

Comment:
comparing floats (Γ[i] is a float) is dangerous, should we use `jnp.allclose(Γ[i], 1)`?

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/62/commits/48d99f3d94274d38bf8fcc436ee278738e2ca6dc